src/main/java/io/github/mzmine/datamodel/features/types/graphicalnodes/BufferedChartNode.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2004-2022 The MZmine Development Team
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.mzmine.datamodel.features.types.graphicalnodes;
+
+import io.github.mzmine.main.MZmineCore;
+import java.awt.image.BufferedImage;
+import java.util.logging.Logger;
+import javafx.animation.PauseTransition;
+import javafx.embed.swing.SwingFXUtils;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.BorderPane;
+import javafx.util.Duration;
+import org.jfree.chart.fx.ChartViewer;
+
+/**
+ * Used to buffer charts with ImageView and predefined sizes. Can save performance in tables
+ */
+public class BufferedChartNode extends BorderPane {
+
+  private static final Logger logger = Logger.getLogger(BufferedChartNode.class.getName());
+  private ChartViewer chart;
+  private ImageView imageView;
+  private boolean makeInteractiveOnClick;
+
+  public BufferedChartNode(final boolean makeInteractiveOnClick) {
+    this.makeInteractiveOnClick = makeInteractiveOnClick;
+    PauseTransition resizer = new PauseTransition(Duration.seconds(2));
+
+    widthProperty().addListener((observable, oldValue, newValue) -> {
```

Comment:
I think the nice thing about the image is, is that we can do all the calculations on a seperate thread, including creation of the chart and the image. When the column is changed now, all images are recalcuated on the FX thread, or?

Reply:
They are only recalculated once you scroll down and see the images. We could abstract this part from the jfx thread and then push the change on the JavaFX thread I guess